### PR TITLE
Using SparkContext#registerLineage instead

### PR DESCRIPTION
**Pull request by gokulavasan**
Depends on PR https://github.com/caskdata/cdap/pull/8180

JIRA : https://issues.cask.co/browse/HYDRATOR-1367

### CODE REVIEW

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by gokulavasan** [comment]
Build passed. Will merge after dependent PR in CDAP repo is merged.

**Pull request merged**
